src/autofill-utils.js

```diff
@@ -566,6 +566,35 @@ function getActiveElement (root = document) {
     return innerActiveElement
 }
 
+/**
+ * Takes a root, creates a treewalker and finds all shadow elements that match the selector
+ * @param {HTMLElement} root
+ * @param {string} selector
+ * @returns {HTMLElement[]}
+ */
+function findEnclosedShadowElements (root, selector) {
+    const shadowElements = []
+    const walker = document.createTreeWalker(root, NodeFilter.SHOW_ELEMENT)
+
+    let node = walker.nextNode()
+    while (node) {
+        if (node instanceof HTMLElement && node.shadowRoot) {
+            shadowElements.push(node.shadowRoot.querySelectorAll(selector))
+        }
+        node = walker.nextNode()
+    }
+
+    const elements = []
+    shadowElements.forEach((shadowElementList) => {
+        shadowElementList.forEach((el) => {
+            if (el instanceof HTMLElement) {
+                elements.push(el)
+            }
+        })
+    })
```

Comment:
I'm not quite sure why this is needed. It seems like we're already adding these elements in the previous loop. Can we destructure the lists in the previous loop so we avoid these extra steps?

Reply:
yes, this looks like too much work - push accepts multiple arguments, so you could just do this instead.

```js
shadowElements.push(...node.shadowRoot.querySelectorAll(selector))
```

Reply:
Had done a super quick version at the beginning and then I moved on, and forgot about changing it :D 

Fixed here: https://github.com/duckduckgo/duckduckgo-autofill/pull/592/commits/5ec6e2a4503675df86d0ad0db9fd5d783f36ff4a